libs/core/src/scss/interaction-state/_hover.scss

```diff
@@ -0,0 +1,54 @@
+@use 'sass:color';
+@use 'sass:meta';
+@use '../utils';
+
+// TODO: Accessibility - Loudness must respect user preferences
+// See https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-contrast
+// TODO: Await UX decision on default loudness and update
+$default-loudness: utils.get-loudness('loud');
+
+@mixin apply-hover($background-color, $color, $property, $loudness: null) {
+  @if meta.type-of($background-color) != color {
+    @error "$background-color must be a valid CSS color value. See https://developer.mozilla.org/en-US/docs/Web/CSS/color_value";
+  }
+
+  @if meta.type-of($color) != color {
+    @error "$color must be a valid CSS color value. See https://developer.mozilla.org/en-US/docs/Web/CSS/color_value";
+  }
+
+  // ---------------------------------------------------------------------------
+  // Determine if lightness should increase or decrease.
+  // The default is increase (a positive percentage).
+  //
+  // When background color is transparent assume that the canvas underneath is
+  // a light color. We cannot really determine if lightness should increase or
+  // decrease, but this is a best guess. It's possible to work around by passing
+  // a $background-color (or $color) parameter with a value other than
+  // transparent.
+  //
+  // Sidenote: The color transparent is defined as black with 0% opacity, i.e.,
+  // 100% transparent. Often declared as rgba(0, 0, 0, 0%) in browsers.
+  // ---------------------------------------------------------------------------
+  $relative-lightness: $default-loudness !default;
+
+  @if $loudness {
+    $relative-lightness: $loudness;
+  }
+
+  @if color.lightness($background-color) > color.lightness($color) {
+    // Decrease relative lightness
+    $relative-lightness: $relative-lightness * -1;
+  }
+
+  cursor: pointer;
+  transition-property: background-color;
+  transition-duration: utils.get-transition-duration('quick');
+  transition-timing-function: utils.get-transition-easing('static');
+
+  @include utils.hover() {
+    // TODO: Accessibility - check that contrast is (still) sufficiently high
```

Comment:
TODO should be handled before we merge 😄

Reply:
Issue created: #2014